CMakeLists.txt

```diff
@@ -21,20 +24,96 @@ set(PUNICA_FULL_VERSION "${PROJECT_NAME} ${PUNICA_VERSION}")
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/src/version.h)
 
+list(APPEND CMAKE_MODULE_PATH ${ULFIUS_CMAKE_MODULES_DIR})
+
 include(${CMAKE_CURRENT_LIST_DIR}/src/punica.cmake)
 
 add_executable(${PROJECT_NAME} ${PUNICA_SOURCES})
-find_library(ULFIUS_LIB ulfius)
-find_library(JANSSON_LIB jansson)
-find_library(JWT_LIB jwt)
-find_library(WAKAAMA_LIB wakaama)
-find_library(ORCANIA_LIB orcania)
-target_compile_options(${PROJECT_NAME} PRIVATE "-Wall" "-pthread")
-target_link_libraries(${PROJECT_NAME} pthread "${ULFIUS_LIB}" "${JANSSON_LIB}" "${JWT_LIB}" "${WAKAAMA_LIB}" "${ORCANIA_LIB}")
 
+# Find SHARED libraries
+set(CMAKE_FIND_LIBRARY_SUFFIXES
+    ${CMAKE_SHARED_LIBRARY_SUFFIX}
+    ${CMAKE_SHARED_MODULE_SUFFIX}
+    ${CMAKE_STATIC_LIBRARY_SUFFIX}
+    )
+
+# Nettle and Hogweed comes together, it depends on gmp
```

Comment:
Is this still needed since we're not building Gnu TLS?

Reply:
Yep, at least my setup failed to build without specifying to link GnuTLS dependencies. I can look into it if it looks "_workaround-ish_".